File: Java/src/main/java/QuestionBank/Solution401_500/Solution404.java
package QuestionBank.Solution401_500;

import tool.TreeNode;

/**
 * @author windows
 */
public class Solution404 {
    int sum = 0;
    public int sumOfLeftLeaves(TreeNode root) {
        process(root);
        return sum;
    }

    private void process(TreeNode root) {
        if (root == null) {
            return;
        }
        if (root.left != null && root.left.left == null && root.left.right == null) {
            sum += root.left.val;
        }
        process(root.left);
        process(root.right);
    }


    public static void main(String[] args) {
        Solution404 solution404 = new Solution404();
    }
}

File: C++/src/Solution401_500/Solution404.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution404.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;


//Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};


int sumOfLeftLeaves(TreeNode* root) {
    if (root == nullptr){
      return 0;
    }
    int sum = 0;
    if (root->left != nullptr && root->left->left == nullptr && root->left->right == nullptr){
      sum += root->left->val;
    }
    return sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right) + sum;
}